pkg/config/config.go

```diff
@@ -0,0 +1,29 @@
+package config
+
+import (
+	"context"
+
+	"github.com/sethvargo/go-envconfig"
+)
+
+type Configuration struct {
+	DevfileProxy          *string `env:"DEVFILE_PROXY,noinit"`
+	DockerCmd             string  `env:"DOCKER_CMD,default=docker"`
+	Globalodoconfig       *string `env:"GLOBALODOCONFIG,noinit"`
+	OdoDebugTelemetryFile *string `env:"ODO_DEBUG_TELEMETRY_FILE,noinit"`
+	OdoDisableTelemetry   *bool   `env:"ODO_DISABLE_TELEMETRY,noinit"`
+	OdoLogLevel           *int    `env:"ODO_LOG_LEVEL,noinit"`
+	OdoTrackingConsent    *string `env:"ODO_TRACKING_CONSENT,noinit"`
+	PodmanCmd             string  `env:"PODMAN_CMD,default=podman"`
+	TelemetryCaller       string  `env:"TELEMETRY_CALLER,default="`
+	OdoExperimentalMode   bool    `env:"ODO_EXPERIMENTAL_MODE,default=false"`
```

Comment:
One comment on this. So it means that setting `ODO_EXPERIMENTAL_MODE` to a non-parseable value would return an error:

```
$ ODO_EXPERIMENTAL_MODE=foo odo dev --run-on=podman
 ✗  OdoExperimentalMode("foo"): strconv.ParseBool: parsing "foo": invalid syntax
```

In terms of user experience (https://github.com/redhat-developer/odo/issues/6150), I think we could be liberal in what we accept here and just run odo in non-experimental mode, no?

Reply:
I'm not convinced. I would prefer to have a frank error explaining the value for the variable is not valid. 

In the other case, the user will get an error that the `--run-on` flag fdoes not exist, and it won't be obvious it is because of a wrong value on the variable

Reply:
That makes sense. As discussed together yesterday, we therefore need to update the documentation of `ODO_EXPERIMENTAL_MODE` environment variable to list [all possible values](https://pkg.go.dev/strconv#ParseBool): https://github.com/redhat-developer/odo/blob/main/docs/website/docs/overview/configure.md?plain=1#L193

Reply:
Done in https://github.com/redhat-developer/odo/pull/6293/commits/e0a7f88638885bac8b31ff47e670fc35a2cd3326